libraries/AP_InertialSensor/AP_InertialSensor_Invensense.cpp

```diff
@@ -500,6 +500,16 @@ bool AP_InertialSensor_Invensense::_accumulate_sensor_rate_sampling(uint8_t *sam
         Vector3f g2 = g * _gyro_scale;
         _notify_new_gyro_sensor_rate_sample(_gyro_instance, g2);
 
+        const int16_t overflowValue = 0x7C00; // 1937 dps
+        if (g.x > overflowValue || g.x < -overflowValue
```

Comment:
the g variable is in radians/sec at this point. Comparing to 0x7c00 doesn't make much sense

Reply:
surely _gyro_scale converts the 16bit value to radians?

Reply:
I definitely think you are wrong :)
_gyro_scale is radians(1)/16.4. The sensor maximum output is 32767 (signed short), 32767/16.4 ~2000 so _gyro_scale converts g (scaled dps) to g2 (radians ps)